src/drivers/display/nxp_elcdif/lv_nxp_elcdif.c

```diff
@@ -0,0 +1,168 @@
+/**
+ * @file lv_nxp_elcdif.c
+ *
+ * Driver for NXP's ELCD
+ */
+
+#include "lv_nxp_elcdif.h"
+
+#if LV_USE_NXP_ELCDIF == 1
+
+lv_indev_t * lv_nxp_indev_create(void (indev_init_cb)(void), lv_indev_type_t indev_type, lv_indev_read_cb_t read_cb)
+{
+    lv_indev_t * indev;
+
+    indev = lv_indev_create();
+    if(!indev) {
+        LV_LOG_ERROR("Malloc failed: lv_nxp_indev_create")
+        return NULL;
+    }
+    lv_indev_set_type(indev, indev_type);
+
+    /*Set callbacks*/
+    if(read_cb)
+        lv_indev_set_read_cb(indev, read_cb);
+
+    if(indev_init_cb)
+        indev_init_cb();
+
+    return indev;
+}
+
+lv_indev_t * lv_nxp_touchpad_create(void (touchpad_init_cb)(void), lv_indev_read_cb_t touchpad_read_cb)
+{
+    /*Handle to touchpad*/
+    lv_indev_t * touchpad;
+
+    touchpad = lv_nxp_indev_create(touchpad_init_cb, LV_INDEV_TYPE_POINTER, touchpad_read_cb);
+    return touchpad;
+}
```

Comment:
`indev`'s are unrelated to ELCDIF and should be removed.

Reply:
My train of thought was that they use the input device in the demo as well, so a simple wrapper could be supported alongside the display drivers.